### PR TITLE
ci(release): add workflow to update identity version

### LINKED ISSUE #12920: Add a github workflow that allows updating the identity version on a specified branch

### PR DESCRIPTION
**Pull request by megglos**
## Description

Allows to trigger an update of the `version.identity` property on an arbitrary targetBranch.
This is supposed to be triggered by a task from the release process going forward to fully automate this step.

In case the version is already the desired (due to manual or dependabot update) the workflow will skip the update.

## Related issues

closes #12920

### CODE REVIEW

**Comment by megglos** [comment]
Test runs:
- [Updating to the version already used](https://github.com/camunda/zeebe/actions/runs/5135943242)
- [Updating to a different version (dry run)](https://github.com/camunda/zeebe/actions/runs/5135944146)
- [Trying to update the version on a branch that has no identity version](https://github.com/camunda/zeebe/actions/runs/5135944509)

**Review by lenaschoenburg — APPROVED** [review]
The workflow itself looks good to me :+1: 

I'd like to see this workflow evolve into a "prepare release" workflow that takes care of creating the release branch, updating identity, starting qa etc.
For now it's fine but I think long-term we should try to not have too many small, specialized workflows because they will become difficult to maintain.

File: .github/workflows/update-identity.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          ref: ${{ inputs.targetBranch }}
+      - name: Git User Setup
+        id: git-setup
+        run: |
+          git config --global user.email "github-actions[update-identity]"
+          git config --global user.name "github-actions[update-identity]@users.noreply.github.com"
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
```
[the remark this answers is not in the document]
Comment (megglos) [reply .github/workflows/update-identity.yml]: I now just thought, why not just using our own `setup-zeebe` action here? this way it's always aligned with other jobs wdyt?

[the remark this answers is not in the document]
Comment (oleschoenburg) [reply .github/workflows/update-identity.yml]: Sure but it feels a bit heavy-handed for what we are actually doing: setting a property. I don't feel strongly about this though, so whatever makes sense to you :+1:

[the remark this answers is not in the document]
Comment (megglos) [reply .github/workflows/update-identity.yml]: thanks to repo caching it's also ~1m faster ^^ https://github.com/camunda/zeebe/actions/runs/5141474200/jobs/9254009264

**Comment by megglos** [comment]
bors merge

**Pull request merged**